### PR TITLE
Update GogsWebHook.java

### PR DESCRIPTION
**Pull request by S3NS4Ti0N**
Added support for multibranch pipelines

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/gogs/GogsWebHook.java [hunk 1]
```diff
@@ -144,7 +144,7 @@ public void doIndex(StaplerRequest req, StaplerResponse rsp)  throws IOException
             ACL acl = jenkins.getACL();
             acl.impersonate(ACL.SYSTEM);
 
-            Job job = GogsUtils.find(jobName, Job.class);
+            Job job = GogsUtils.find(jobName, Job.class); 
 
             if (job != null) {
```
Comment (fabienjuif) [inline src/main/java/org/jenkinsci/plugins/gogs/GogsWebHook.java]: I would try to unify some code : 

```java
// 1. Get the job
Job job = GogsUtils.find(jobName, Job.class);
if (job == null) { /* Here is the @S3NS4Ti0N fix */
  String ref = (String)jsonObject.get("ref"); 
  String[] components = ref.split("/"); 
  ref = components[components.length-1]; 
                
  job = GogsUtils.find(jobName + "/" + ref, Job.class); 
}

// 2. Use the job found
if (job != null) {
  /* FIXME : I don't know the purpose of it
  foundJob = true;
  */

  /* secret is stored in the properties of Job */
  final GogsProjectProperty property = (GogsProjectProperty) job.getProperty(GogsProjectProperty.class);
  if (property != null) { /* only if Gogs secret is defined on the job */
    jSecret = property.getGogsSecret(); /* Secret provided by Jenkins */
  }
}
```

**Comment by pascalw** [comment]
Thanks for the patch @S3NS4Ti0N I was looking for exactly this. Will you make the changes suggested by @fabienjuif? If you want I could also make the changes. Let me know.

File: src/main/java/org/jenkinsci/plugins/gogs/GogsWebHook.java [hunk 2]
```diff
@@ -153,6 +153,23 @@ public void doIndex(StaplerRequest req, StaplerResponse rsp)  throws IOException
 				if (property != null) { /* only if Gogs secret is defined on the job */
 					jSecret = property.getGogsSecret(); /* Secret provided by Jenkins */
 				}
+            } 
+            else {
+                String ref = (String)jsonObject.get("ref"); 
+                String[] components = ref.split("/");
```
Comment (pascalw) [inline src/main/java/org/jenkinsci/plugins/gogs/GogsWebHook.java]: It's pretty common for branches to be prefixed with `feature/` or `bugfix/`. A branch for example named `feature/add-something` would be detected as `add-something` with this code.

Looking at my Gogs webhook logs I see the ref is always in the format `refs/heads/feature/add-something`, so perhaps we could simply use `String branchName = ref.replace("refs/heads/", "");`. What do you think?

**Comment by pascalw** [comment]
@S3NS4Ti0N @fabienjuif I pushed my changes here https://github.com/jenkinsci/gogs-webhook-plugin/compare/master...pascalw:feature/support-multibranch?expand=1 in case you're interested. I also made some additional changes to support branches with `/` chars in the branch name.

**Comment by sanderv32** [comment]
Coming weekend I will test it and merge this PR

**Comment by S3NS4Ti0N** [comment]
Whoop Whoop

**Comment by sanderv32** [comment]
Merged the PR

**Pull request merged**